### DISCUSSION TITLE
scraping error for Azure Database for MySQL flexible server using resource uri

### DISCUSSION THREAD

**Discussion by sumant5476** [tomkerkhove/promitor#2033] [Q&A] [answered]
promitor-agent-scraper-24  | [10:01:33 INF] Booting up Promitor v2.6.0 running .NET .NET 6.0.1 - Thank you for using Promitor!
promitor-agent-scraper-24  | [10:01:33 INF] Running .NET 6.0.1 on Linux (alpine.3.14-x64 | Linux 3.10.0-862.14.4.el7.x86_64 #1 SMP Fri Sep 21 09:07:21 UTC 2018).
promitor-agent-scraper-24  | [10:01:33 INF] Using configuration folder '/config/'
promitor-agent-scraper-24  | [10:01:34 FTL] Promitor Scraper Agent has encountered an unexpected error. Please open an issue at https://github.com/tomkerkhove/promitor/issues to let us know about it.
promitor-agent-scraper-24  | System.ArgumentNullException: [scraping] cannot be Null. (Parameter 'scraping')
promitor-agent-scraper-24  |    at GuardNet.Guard.For[TException](Func`1 predicate, TException exception)
promitor-agent-scraper-24  |    at GuardNet.Guard.NotNull[TParam,TException](TParam param, TException exception)
promitor-agent-scraper-24  |    at GuardNet.Guard.NotNull[TParam](TParam param, String paramName, String message)
promitor-agent-scraper-24  |    at GuardNet.Guard.NotNull[TParam](TParam param, String paramName)
promitor-agent-scraper-24  |    at Promitor.Core.Scraping.Configuration.Model.Metrics.ScrapeDefinition`1..ctor(AzureMetricConfiguration azureMetricConfiguration, PrometheusMetricDefinition prometheusMetricDefinition, Scraping scraping, TResourceDefinition resource, String subscriptionId, String resourceGroupName) in /src/Promitor.Core.Scraping/Model/Metrics/ScrapeDefinition.cs:line 33
promitor-agent-scraper-24  |    at Promitor.Core.Scraping.Configuration.Model.Metrics.MetricDefinition.CreateScrapeDefinition(IAzureResourceDefinition resource, AzureMetadata azureMetadata) in /src/Promitor.Core.Scraping/Model/Metrics/MetricDefinition.cs:line 66
promitor-agent-scraper-24  |    at Microsoft.Extensions.DependencyInjection.SchedulingExtensions.ScheduleResourceScraping(IAzureResourceDefinition resource, AzureMetadata azureMetadata, MetricDefinition metric, AzureMonitorClientFactory azureMonitorClientFactory, MetricSinkWriter metricSinkWriter, IAzureScrapingPrometheusMetricsCollector azureScrapingPrometheusMetricCollector, IConfiguration configuration, IOptions`1 azureMonitorLoggingConfiguration, ILoggerFactory loggerFactory, ILogger`1 logger, IServiceCollection services) in /src/Promitor.Agents.Scraper/SchedulingExtensions.cs:line 70
promitor-agent-scraper-24  |    at Microsoft.Extensions.DependencyInjection.SchedulingExtensions.ScheduleMetricScraping(IServiceCollection services) in /src/Promitor.Agents.Scraper/SchedulingExtensions.cs:line 33
promitor-agent-scraper-24  |    at Promitor.Agents.Scraper.Startup.ConfigureServices(IServiceCollection services) in /src/Promitor.Agents.Scraper/Startup.cs:line 54
promitor-agent-scraper-24  |    at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
promitor-agent-scraper-24  |    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
promitor-agent-scraper-24  |    at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.InvokeCore(Object instance, IServiceCollection services)
promitor-agent-scraper-24  |    at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.<>c__DisplayClass9_0.<Invoke>g__Startup|0(IServiceCollection serviceCollection)
promitor-agent-scraper-24  |    at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.Invoke(Object instance, IServiceCollection services)
promitor-agent-scraper-24  |    at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.<>c__DisplayClass8_0.<Build>b__0(IServiceCollection services)
promitor-agent-scraper-24  |    at Microsoft.AspNetCore.Hosting.GenericWebHostBuilder.UseStartup(Type startupType, HostBuilderContext context, IServiceCollection services, Object instance)
promitor-agent-scraper-24  |    at Microsoft.AspNetCore.Hosting.GenericWebHostBuilder.<>c__DisplayClass13_0.<UseStartup>b__0(HostBuilderContext context, IServiceCollection services)
promitor-agent-scraper-24  |    at Microsoft.Extensions.Hosting.HostBuilder.CreateServiceProvider()
promitor-agent-scraper-24  |    at Microsoft.Extensions.Hosting.HostBuilder.Build()
promitor-agent-scraper-24  |    at Promitor.Agents.Scraper.Program.Main(String[] args) in /src/Promitor.Agents.Scraper/Program.cs:line 35

**Comment by tomkerkhove** [comment, marked as answer]
Let's continue on https://github.com/tomkerkhove/promitor/discussions/1999#discussioncomment-2773492 as this is going to be the same issue.
